src/Compiler/Checking/AugmentWithHashCompare.fsi

```diff
@@ -21,7 +21,7 @@ val MakeValsForCompareWithComparerAugmentation: TcGlobals -> TyconRef -> Val
 
 val MakeValsForEqualsAugmentation: TcGlobals -> TyconRef -> Val * Val
 
-val MakeValsForEqualityWithComparerAugmentation: TcGlobals -> TyconRef -> Val * Val * Val
+val MakeValsForEqualityWithComparerAugmentation: TcGlobals -> TyconRef -> Val * Val * Val * Val
```

Comment:
Introducing a 4th unnamed Val within the same quadruple might be a good signal for having this as a dedicated (value) type with proper field names.

Reply:
Based on the name and the types returned, it is not clear which val represents what.

Reply:
Fair enough, refactored to a helper type